### ISSUE TITLE
Not all path return a value issue with multiline code.

### ISSUE THREAD

**Issue by wagfeliz** [godotengine/godot#77277]
### Godot version

4.02.stable

### System information

Linux Ubunto 22.04

### Issue description

The following code is invalid : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; return ""

But this is fine : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; 
	return ""

### Steps to reproduce

The following code is invalid : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; return ""

But this is fine : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; 
	return ""

### Minimal reproduction project

The following code is invalid : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; return ""

But this is fine : 

func get_normaldir( normal:Vector3 ) -> String:
	var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
	if(x>y and x>z): return "x"; if(y>x and y>z): return "y"; if(z>x and z>y): return "z"; 
	return ""

**Comment by wagfeliz** [comment]
The function is bugged by the way.... Its just a example.

**Comment by AThousandShips** [comment]
Just repeating the same in each section isn't helpful, also please format your code snippets they are hard to read like this, use "```gdscript" at the start and "```" at the end as separate rows

**Comment by AThousandShips** [comment]
The reason for this is that you aren't doing what you are thinking you are doing

If statements are indentation based, your code is equivalent to:
```gdscript
func get_normaldir( normal:Vector3 ) -> String:
    var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
    if(x>y and x>z):
        return "x"; 
        if(y>x and y>z):
            return "y";
            if(z>x and z>y):
                return "z";
                return ""
```
*not*
```gdscript
func get_normaldir( normal:Vector3 ) -> String:
    var x:float=abs(normal.x); var y:float=abs(normal.y); var z:float=abs(normal.z);
    if(x>y and x>z):
        return "x"; 
    if(y>x and y>z):
        return "y";
    if(z>x and z>y):
        return "z";
    return ""
```

**Comment by wagfeliz** [comment]
I guess I didnt put an good example. By the way, the reason why I repeated the code on each section was because my browse glitch and didnt allow me to type on it.

If you have a code like : 
if(something): return that else: return this; return somethingelse
You will get an error that its impossible to fix unless you dont use multiline code. You will force do to something like this to fix: 

if(something): return that else: return this; 
return somethingelse

The issue is with the multiline interpreter of the gdscript or with the error handler for if then else, that is not considering multiline code, I am not sure if the same happens with c#.

**Comment by wagfeliz** [comment]
Following some more consistent example:

```
# this is invalid :
func test(a:int) -> int:
	if(a==1): return 0; if(a==0): return 1; return -1

# this is valid but should be the same as above :
func test2(a:int) -> int:
   if(a==1): return 0; if(a==0): return 1; 
   return -1
```

As I understand for the multiline code convention, it should work, so I think its a bug.

**Comment by wagfeliz** [comment]
> The reason for this is that you aren't doing what you are thinking you are doing, not a bug, and if you check code warnings you will see that the code is unreachable, indicating your error
> 
> If statements are indentation based, your code is equivalent to:
> 
> ```gdscript
> func get_normaldir( normal:Vector3 ) -> String:
>     var x:float=abs(normal.x)
>     var y:float=abs(normal.y)
>     var z:float=abs(normal.z)
>     if(x>y and x>z):
>         return "x"
>         if(y>x and y>z):
>             return "y"
>             if(z>x and z>y):
>                 return "z"
>                 return ""
> ```
> 
> _not_
> 
> ```gdscript
> func get_normaldir( normal:Vector3 ) -> String:
>     var x:float=abs(normal.x)
>     var y:float=abs(normal.y)
>     var z:float=abs(normal.z)
>     if(x>y and x>z):
>         return "x"
>     if(y>x and y>z):
>         return "y"
>     if(z>x and z>y):
>         return "z"
>     return ""
> ```
> 
> Multiple statements on the same line is [not recommended](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#one-statement-per-line)

I dont agree... My understand of multiline convention is that when you use an  endstatement the interpreter/compiler should consider same as an new line, eater that or it makes an dangerous mess in the language.

**Comment by AThousandShips** [comment]
> I dont agree

Agree with what? This *is* how the code works, not an opinion

**Comment by wagfeliz** [comment]
> > I dont agree
> 
> Agree with what? This _is_ how the code works, not an opinion

Do you know what is the function of the ";" char in the language ?

**Comment by AThousandShips** [comment]
To end a statement, not to end a block

**Comment by AThousandShips** [comment]
GDScript is an indentation based language, this is how it works

So to give you an example:
```gdscript
if a:
    if b:
       if c: foo(); bar()
```

Should this be read as:
```gdscript
if a:
    if b:
       if c:
           foo()
           bar()
```
or
```gdscript
if a:
    if b:
       if c:
           foo()
       bar()
```
or
```gdscript
if a:
    if b:
       if c:
           foo()
   bar()
```
or
```gdscript
if a:
    if b:
       if c:
           foo()
bar()
```

**Comment by AThousandShips** [comment]
> As I understand for the multiline code convention

What convention, where is this documented?

**Comment by AThousandShips** [comment]
My issue here is that I don't think:
```gdscript
if x: foo(); bar()
```
Should be different from
```gdscript
if x:
    foo(); bar
```
It'd be very confusing

I am sorry if I am being dismissive, but this is a matter of how the language is designed, and to me makes the most sense and follows the principle of least astonishment, the base assumption would be "this doesn't imply that we change indentation"

**Comment by AThousandShips** [comment]
This was already discussed in godotengine/godot-proposals#3759, see https://github.com/godotengine/godot-proposals/issues/3759#issuecomment-1005974260

**Comment by AThousandShips** [comment]
Opened a proposal for the discontinuing of this code, feel free to voice your opinion, concern, and reasoning there godotengine/godot-proposals#6924

Otherwise I'd consider this not a bug and suggest further conversation at the proposal

**Comment by HolonProduction** [comment]
Closing in favor of the linked proposal.
